cs/src/core/Allocator/AllocatorBase.cs

```diff
@@ -1471,9 +1473,14 @@ private void AsyncGetFromDiskCallback(uint errorCode, uint numBytes, NativeOverl
                     // We have the complete record.
                     if (RetrievedFullRecord(record, ref ctx))
                     {
-                        if (comparer.Equals(ref ctx.request_key.Get(), ref GetContextRecordKey(ref ctx)))
+                        // If the keys are same, then I/O is complete. For PSFs, we may be querying on an address instead of a key;
+                        // in this case, ctx.request_key is null for the primary FKV but we know the address is the one we want.
+                        var isEqualKey = ctx.request_key is null 
```

Comment:
This can use the new address-based variant of Read, instead of key-based variant of Read. Then no changes are needed.